worker/worker.go

```diff
@@ -173,9 +165,28 @@ func detectFeatures(name string, data map[string][]byte, namespace *database.Nam
 		return
 	}
 
-	// Ensure that every feature has a Namespace associated, otherwise associate the detected
-	// namespace. If there is no detected namespace, we'll throw an error.
+	// If there are no feature detected, use parent's features if possible.
+	// TODO(Quentin-M): We eventually want to give the choice to each detectors to use none/some
+	// parent's Features. It would be useful for detectors that can't find their entire result using
+	// one Layer.
+	if len(features) == 0 && parent != nil {
+		features = parent.Features
+		return
+	}
+
 	for i := 0; i < len(features); i++ {
+		// If the FeatureVersion (name/version) is present in both the parent and the current layer,
+		// but the Namespaces are different (e.g. dist-upgrade), we should use the parent's Namespace.
+		if parent != nil && namespace != parent.Namespace {
+			for _, parentFeatureVersion := range parent.Features {
```

Comment:
Let's build a map of the parent's features once before the `for i` loop and then just look up each iteration.